### DISCUSSION TITLE
OpenVAS CSV Parser. Findings not linked to endpoints and CVE not displayed in the vulnerability ID

### DISCUSSION THREAD

**Discussion by GilFernandes2000** [DefectDojo/django-DefectDojo#9780] [Q&A] [answered]
Hello, I hope everybody is doing great.

I'm integrating OpenVAS scan results into DefectDojo using the CSV format.
It imports the results as expected, but I noticed that the data is not processed as some other scan imports.
What I mean is the data about the vulnerability is all there, but it misses some things such as linking the finding to the proper endpoint and the vulnerability ID (CVE) not applied on the finding preview. 
I'll give some examples

![Captura de ecrã 2024-03-18 223906](https://github.com/DefectDojo/django-DefectDojo/assets/82401775/b3d92bd7-97e1-46e2-b79d-ecc3d205f3d9)

![Captura de ecrã 2024-03-18 224038](https://github.com/DefectDojo/django-DefectDojo/assets/82401775/c1d54854-355a-493b-b4a0-dbb798f46267)

In these images, we can see the same finding originated from different scanners. Nmap and OpenVAS respectively. As you can see, in the Nmap scan, the cve is displayed on the Vulnerability ID with a direct link to the CVE on the NVD. The same finding originated from an OpenVAS scan, this does not show.

Another thing.

![Captura de ecrã 2024-03-18 224734](https://github.com/DefectDojo/django-DefectDojo/assets/82401775/48851b3f-1300-4c74-95ed-b17e0c374e4f)

![Captura de ecrã 2024-03-18 224848](https://github.com/DefectDojo/django-DefectDojo/assets/82401775/a67dc644-adc7-425a-a156-8b6bdfb025bf)

Here, we see the same finding from above. As you can see the link on the endpoint from the Nmap Scan is rightly linked and it forwards to the endpoint where we can see more findings associated with it. In the OpenVAS scan doesn't link to the correct endpoint(missing the port which is found in the description), and we can't access these findings from the endpoint's panel.

Can these features be implemented? I would appreciate it very much.

Thank you all in advance.

**Comment by manuel-sommer** [comment, marked as answer]
Hi @GilFernandes2000,

first of all, thank you for raising this issue. You could have opened up an issue / bug report to fix this issue rather than a discussion, but no worries. I made a PR for this: https://github.com/DefectDojo/django-DefectDojo/pull/9791 
Regarding the first mentioned problem: The CVE column, I adapted the code and it fixes the reported issue. In case there are multiple CVEs, the rest of the cves are included in the description. 
Regarding the endpoint port problem, the functionality is already given, as the code considers the "port column" (see unittests). I can try to advance the code in a way that it uses the description in case the port column is empty. For that, I would need more csv examples and I can't guarantee that this is actually duable as it is not a straight forward fix, because the description contains a lot of variable content. This is kind of an edgecase as it looks like. 

Best,

Manuel

**Reply by GilFernandes2000** [reply]
Sure, here are the csv reports from both types of openVAS scanners (openVAS and CVE)
[report_openvas.zip](https://github.com/DefectDojo/django-DefectDojo/files/14665224/report_openvas.zip)
